core/embed/boardloader/main.c

```diff
@@ -327,7 +297,6 @@ int main(void) {
        check_image_contents(old_hdr, IMAGE_HEADER_SIZE, &BOOTLOADER_AREA))) {
     write_bootloader_min_version(old_hdr->monotonic);
```

Comment:
Maybe we should avoid updating the monotonic counter here and instead do it later, just before jumping to the bootloader, when the bootloader has been fully checked. Updating the counter prematurely could potentially prevent downgrading to a previous (but functional) version.

Reply:
This update is done with version of currently installed bootloader (not the one being installed from SD card. The checks (both sig and contents) are done before updating. 

I need to do that here - if bootloader has just been updated from fw, user has inserted an SD card with older version and reboots - then not updating the counter here would lead to possibility of downgrade below the currently installed version.

Reply:
Okay, I see it, thanks for explanation.